src/commands/dry_run.rs

```diff
@@ -96,17 +97,34 @@ where
 		raw_solution.encode().len(),
 	);
 
-	let tx = epm::signed_solution(raw_solution)?;
-	let xt = api
-		.tx()
-		.create_signed_with_nonce(&tx, &*signer, nonce, ExtrinsicParams::default())?;
-
-	let outcome = api.rpc().dry_run(xt.encoded(), config.at).await?;
-
-	log::info!(target: LOG_TARGET, "dry-run outcome is {:?}", outcome);
-
-	match outcome {
-		Ok(()) => Ok(()),
-		Err(e) => Err(Error::Other(format!("{e:?}"))),
+	// Now we've logged the score, check whether the solution makes sense:
+	miner_solution.feasibility_check()?;
```

Comment:
I realized that this fail when `--force-winner-count` is enabled because it would fetch `MinerConfig::MaxWinners` in check and always fails.

We can remove this completely or overwrite `MinerConfig::MaxWinners`

Reply:
Yup, it would!

I move it to after the logging for this reason. I figured that when doing a dry run, you might be interested to know about this failure either way (maybe you set `--force-winner-count` to the number of winners you expect the chain to be happy with or osmething).

I did also ponder having it in the `if let Some` block below; that way, we would only fail if we were otherwise already going to try and submit the thing for the node to check. But it felt like one might want to know this regardless

Reply:
as we read the `MaxWinners` from the metadata and one would provide not use the same values in the metadata that will always fails.

so either just do the feasibility_check if `force_winner_count.is_some()` or add this in the beginning of this function:

```rust
	if Some(max_winners) = config.force_winner_count {
		static_types::MaxWinners.set(max_winners);
	}
```

Reply:
Okeydoke; I disabled if `.is_none()`!